core/module.h

```diff
@@ -132,7 +97,8 @@ struct Command {
 
 using Commands = std::vector<struct Command>;
 
-// A class for generating new Modules of a particular type.
+// A class for managing modules of 'a particular type'.
+// It creates new modules and forwarding module-specific commands
```

Comment:
Minor: perhaps rewrite as `Creates new modules and forwards module-specific commands.`